conftest.py

```diff
@@ -378,3 +385,193 @@ def pytest_sessionfinish(session, exitstatus):
+        value=f"{context.appsec_rules_file}",
+    )
+    ET.SubElement(
+        testsuite_props,
+        "property",
+        name="dd_tags[systest.suite.context.scenario]",
+        value=f"{os.environ.get('SYSTEMTESTS_SCENARIO')}",
+    )
+
+
+def _create_junit_testsuite_summary(testsuite_props, summary_json):
+    if "passed" in summary_json:
+        ET.SubElement(
+            testsuite_props,
+            "property",
+            name="dd_tags[systest.suite.summary.passed]",
+            value=f"{ summary_json['passed']}",
+        )
+    if "xfail" in summary_json:
+        ET.SubElement(
+            testsuite_props, "property", name="dd_tags[systest.suite.summary.xfail]", value=f"{ summary_json['xfail']}"
+        )
+    if "skipped" in summary_json:
+        ET.SubElement(
+            testsuite_props,
+            "property",
+            name="dd_tags[systest.suite.summary.skipped]",
+            value=f"{ summary_json['skipped']}",
+        )
+    ET.SubElement(
+        testsuite_props, "property", name="dd_tags[systest.suite.summary.total]", value=f"{ summary_json['total']}"
+    )
+    ET.SubElement(
+        testsuite_props,
+        "property",
+        name="dd_tags[systest.suite.summary.collected]",
+        value=f"{ summary_json['collected']}",
+    )
+
+
+def _create_testcase_results(
+    junit_xml_root, testclass_name, testcase_name, outcome, skip_reason, test_doc, test_rfc, test_coverage, test_release
+):
+
+    testcase = junit_xml_root.find(f"testsuite/testcase[@classname='{testclass_name}'][@name='{testcase_name}']")
+    if testcase:
+        # Change name att because CI Visibility uses identifier: testsuite+name
+        testcase.set("name", testclass_name + "." + testcase_name)
+        # Add custom tags
+        tc_props = ET.SubElement(testcase, "properties")
+        ET.SubElement(tc_props, "property", name="dd_tags[systest.case.outcome]", value=f"{outcome}")
+        ET.SubElement(tc_props, "property", name="dd_tags[systest.case.skip_reason]", value=f"{skip_reason}")
+        ET.SubElement(tc_props, "property", name="dd_tags[systest.case.doc]", value=f"{test_doc}")
+        ET.SubElement(tc_props, "property", name="dd_tags[systest.case.rfc]", value=f"{test_rfc}")
+        ET.SubElement(tc_props, "property", name="dd_tags[systest.case.coverage]", value=f"{test_coverage}")
+        if test_release:
+            for library_name in test_release:
+                ET.SubElement(
+                    tc_props,
+                    "property",
+                    name=f"dd_tags[systest.case.release.library.{library_name}]",
+                    value=f"{test_release[library_name]}",
+                )
+        if outcome == "failed":
+            if bool(skip_reason):
+                tc_skipped = ET.SubElement(testcase, "skipped")
+                tc_skipped.text = skip_reason
+            else:
+                tc_failure = ET.SubElement(testcase, "failure")
+                tc_failure.text = "failed test"
```

Comment:
TODO Get details of the failure

Reply:
if you want to do so, you'll need to modify the code from 

```
data -> json -> xml
```

to 

```
get_data() -> json
get_data() -> xml
```